plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -295,12 +294,12 @@ public List<AddFileEntry> getActiveFiles(TableSnapshot tableSnapshot, ConnectorS
         }
     }
 
-    private List<AddFileEntry> loadActiveFiles(TableSnapshot tableSnapshot, ConnectorSession session)
+    private List<AddFileEntry> loadActiveFiles(TableSnapshot tableSnapshot, MetadataEntry metadataEntry, ProtocolEntry protocolEntry, ConnectorSession session)
     {
-        try (Stream<AddFileEntry> entries = getEntries(
+        try (Stream<AddFileEntry> entries = getAddFileEntries(
                 tableSnapshot,
-                ImmutableSet.of(ADD),
```

Comment:
remove 

```
if (entryTypes.contains(ADD)) {
            metadataAndProtocol = Optional.of(getCheckpointMetadataAndProtocolEntries(
                    session,
                    checkpointSchemaManager,
                    typeManager,
                    fileSystem,
                    stats,
                    checkpoint));
```

from `io.trino.plugin.deltalake.transactionlog.TableSnapshot#getCheckpointTransactionLogEntries`

it's now dead code

Reply:
Not sure about this:

https://github.com/trinodb/trino/blob/b7d7ba8eb9018b308994a55448be8e71915c371f/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriterManager.java#L116-L123

Reply:
the linked code place already reads metadata entry.
let it read protocol as well and pass both